.yarn/sdks/typescript/lib/tsserver.js

```diff
@@ -91,9 +91,25 @@ const moduleWrapper = tsserver => {
   }
 
   function fromEditorPath(str) {
-    return process.platform === `win32`
-      ? str.replace(/^\^?zip:\//, ``)
-      : str.replace(/^\^?zip:/, ``);
+    switch (hostInfo) {
+      case `coc-nvim`:
+        str = str.replace(/\.zip::/, `.zip/`);
+        // The path for coc-nvim is in format of /<pwd>/zipfile:/<pwd>/.yarn/...
+        // So in order to convert it back, we use .* to match all the thing
+        // before `zipfile:`
+        return process.platform === `win32`
+          ? str.replace(/^.*zipfile:\//, ``)
+          : str.replace(/^.*zipfile:/, ``);
+      case `neovim`:
+        str = str.replace(/\.zip::/, `.zip/`);
+        return process.platform === `win32`
+          ? str.replace(/^\^?zipfile:\//, ``)
+          : str.replace(/^\^?zipfile:/, ``);
+      default:
+        return process.platform === `win32`
+          ? str.replace(/^\^?zip:\//, ``)
```

Comment:
Just a thought but I wonder if it would be worth unpacking this a little and having a case for `vscode` so `fromEditorPath` mirrors `toEditorPath`, would perhaps make the intent a little clearer.

From where it was introduced it looks like the prepended `^` is a requirement of `vscode`
https://github.com/yarnpkg/berry/pull/2136/files#diff-2dde4b1afd9df91455f77a86893b55861d1be4e6ea8bcd831c827b0321fb76bdR36